agent/wsclient/client.go

```diff
@@ -285,3 +279,72 @@ func (cs *ClientServerImpl) handleMessage(data []byte) {
 		log.Info("No handler for message type", "type", typeStr)
 	}
 }
+
+func (cs *ClientServerImpl) connectWithProxyIfNeeded(parsedURL *url.URL, request *http.Request) (*tls.Conn, error) {
```

Comment:
I'd rather have this be named something like `websocketConn` and have it return a net.Conn interface. Minor nit though.